sdks/python/apache_beam/transforms/core.py

```diff
@@ -320,6 +320,22 @@ def split_and_size(self, element, restriction):
     for part in self.split(element, restriction):
       yield part, self.restriction_size(element, part)
 
+  def truncate(self, element, restriction):
+    """Truncate the given restriction into finite amount of work when the
+    pipeline starts to drain.
+
+    By default, if the restriction is bounded, it will return the entire current
+    restriction. If the restriction is unbounded, it will return None.
```

Comment:
```suggestion
    By default, if the restriction is bounded, it will return the entire 
    restriction. If the restriction is unbounded, it will not return anything.
```